src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -89,6 +90,55 @@ accountFromString(AccountID& result, std::string const& strIdent, bool bStrict)
         return Json::objectValue;
 }
 
+std::uint64_t
+getStartHint(std::shared_ptr<SLE const> const& sle, AccountID const& accountID)
+{
+    if (sle->getType() == ltRIPPLE_STATE)
+    {
+        if (sle->getFieldAmount(sfLowLimit).getIssuer() == accountID)
+            return sle->getFieldU64(sfLowNode);
+        else if (sle->getFieldAmount(sfHighLimit).getIssuer() == accountID)
+            return sle->getFieldU64(sfHighNode);
+    }
+
+    return sle->getFieldU64(sfOwnerNode);
```

Comment:
Not all ledger entries have an `sfOwnerNode` field.  I'd be more comfortable if you checked `isFieldPresent()` before calling `getFieldU64()`.  Or, alternatively, you could use the `at(~asOwnerNode)` trick to have an optional returned.

With `getFieldU64()` if the field is missing you'll get an exception, which I suppose is not the end of the world.  It just seems prudent to check for the field presence first.

Reply:
Sorry, I was sloppy.  If you choose to use the `~` trick it would look like this:
```
    return (sle->at(~sfOwnerNode).value_or(0ul));
```
If the `sfOwnerNode` is missing you'll end up returning zero.  And this one is exercised by unit tests, so I've tested it.